test/Button.spec.js

```diff
@@ -125,4 +125,20 @@ describe('Function that creates and renders button', () => {
     });
     expect(testButton.classList.contains('button--incorrectAnswer')).toBe(true);
   });
+
+  it('Should create button without class "button--incorrectAnswer" when "isIncorrectAnswer" is "true"', () => {
+    const testButton = Button({
+      id: 'testId',
+      btnText: 'Test',
+      classList: [],
+      onClickFn: undefined,
+      icon: 'fame',
+      isSpecial: false,
+      isCorrectAnswer: false,
```

Comment:
W ogóle nie widzę zmian dla tego, tylko test?
O ile isSpecial jeszcze przepuściłem to tego jednak za żadne skarby świata :) 
Wchodzi nam tutaj logika quizu, a to po prostu zwykły button.
Nie powinien być świadomy, że coś takiego istnieje jak quiz i dobra czy zła odpowiedź. 
Quiz i odpowiedzi na nim jak to się mówi - mają "wyższy poziom abstrakcji". 
Quiz wie o buttonach z odpowiedziami, ale button nic nie powinny wiedzieć o odpowiedziach. 

Zaczęło się niewinne - ze special, ale właśnie o tym mówiłem :) 
Tutaj już musimy to wywalić z Buttona.

Może być komponent np. 

```
const CorrectAnswerButton = ({answerText}) => {
const button = Button(tutaj jakies properties odpowiednie);
return button;
}

const IncorrectAnswerButton = ({answerText}) => {
/*tutaj podobnie*/
}
```

Wykorzeńmy tę praktykę dodawania propertisów jak najszybciej :) Tak robią na tutorialach a niektórzy nawet po 10 latach programowania - ale to nie jest utrzymywalny i rozszerzalny kod.

Reply:
Jest to zabezpieczone. Ale widzę o co Tobie chodzi.
Jeśli chodzi o buttony, to po udzieleniu odpowiedzi button miał zmieniać kolor na zielony lub czerwony. I tutalj to robi.

Reply:
DODATKOWO:
W powyższym przykladnie nie musimy renderować ponownie przycisków. Wystarczy zmiana klasy/tekstu po udzieleniu odpowiedzi i po przejściu do nastepnego pytania.

Reply:
Czekaj, to coś chyba pokręciłeś :P Nie widziałem PRa, który wszedł w międzyczasie. 
Opis testu zobacz - zdaje mi się, że chcesz przetestować changeText a opis mowi co innego :) Dobra, ja to akceptuje.

Reply:
Poprawię opisy.
Dodam jeszcze zmianę klasy!

Reply:
Mam taką prośbę, aby się z tym wstrzymać do mojego commita, bo to czy odpowiedź jest correct czy nie, i jak to wpływa na buttona, jest w wymaganiach do mojego taska i fajnie byłoby gdybym nie musiała jeszcze raz tego zmieniać przez drobny improvement ;)

Reply:
Hej, ale jedno z drugim się nie kłuci nawet jakby Moje/Twoje miało nie być używane

Reply:
@lamparina Z tego co widze to juz jest zmergowane. Wiec i tak bedziesz musiala zmienic. Tutaj jest tylko test z ustawieniem propertiesow. Moze to zmergujmy a najwyzej wprowadzisz zmiane :) ?

Reply:
Jest zmergowane, bo weszło niekontrolowanie. Zrobiłam to trochę inaczej, wobec czego usunęłam fragmenty dotyczące isCorrectAnswer i isIncorrectAnswer, więc trochę bez sensu dodawać tu testy, które zaraz wylecą :)

Reply:
Ania, to co dodałem służy tylko do stylowania i nie wpływa na inne rzeczy i logikę

Reply:
Jeśli nie wywołasz danej metody (lub nie ustawisz specjalnie tak) to nie zostanie przycisk zmieniony.

Reply:
Test jest na changeText tylk opis był mylący. To jest ten problem o którym mówiłem. Lepiej nie zmieniać już istniejącego kodu - button zostawić w spokoju i porobić ta kompozycje - nowe komponenty używające button. Wtedy tych problemów nie ma :) @PiotrWR te rzeczy które mówię, choć zdaje się że nie trzeba bo inaczej działa to jednak się odbijają - na rozbudowie kodu szczególnie. Nawet nie wszystkie argumenty pamiętam w danym momencie dlaczego tak, ale po latach widzę że warto :D Buttonik wspólny sam w sobie zrobiłeś super i niepotrzebnie go zanieczyszczać tymi kwestiami poprawności czy "specjalności".

Dogadajcie się razem jakoś. Ja bym to zmergowal, żeby była zmiana na dev z changeText a najwyżej Ania usuniesz co nie potrzeba.

Reply:
Chodzi mi o to, że dodałeś TC: 'Should create button without class "button--incorrectAnswer" when "isIncorrectAnswer" is "true"', który nie ma racji bytu, bo 'isIncorrectAnswer' usuwam z buttona, gdyż jest tam zbędny. 

Rozumiem, co ten test robi. Sugeruję tylko zmienić nazwę i przerobić najlepiej jakiś inny case, w którym nie ma nic na temat poprawności odpowiedzi. Może jak wrzucę zaraz swoje zmiany, wszystko stanie się jasne. Piszę o tym tylko po to, bo już trochę się naprzerabiałam tych zmian i bez sensu jest dokładanie kolejnych.

Reply:
Hej, to może nie usuwaj jeśli nie musisz. Jeśli źle nazwałem z chęcią poprawię nazwę

Reply:
Dogadajcie to miedzy sobą jak programiści w zespole :D

Reply:
Dogadamy ;) dyskusja jest :D

Reply:
Zobacz sobie PR "Fixed Btn"
To powinno Ciebie zadowolić
<3

Reply:
NAPRAWIONE ;)